### ISSUE TITLE
Inconsistent Escape handling with nested focus-trapping components

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-components#13225]
**monday.com sync:** #18303762159

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Which Component

Popover, Dialog, Input Date Picker, Input Time Picker

### Actual Behavior

Pressing Escape in certain focus-trapping component hierarchies will cause more than one component to close per single Escape.

### Expected Behavior

Pressing Escape in certain focus-trapped component hierarchies can cause multiple components to close from a single key press.

### Reproduction Sample

https://codepen.io/jcfranco/pen/GgoXoOp, https://codepen.io/jcfranco/pen/bNExVKb?editors=1000

### Reproduction Steps

1. Choose input picker type (date or time)
2. Click on launch dialog stack
3. Wait for components to finish opening (picker should be last)
4. Press Escape per active focus-trap
5. Notice both popover and dialog close after the picker is closed (date-picker might require an additional Escape)

### Reproduction Version

3.0.0

### Relevant Info

This might be caused by `focus-trap` handling `Escape` at the document level and us having non-focus-trapping components that handle `Escape` at the component level (e.g,. panel within `dialog`).

### Priority impact

impact - p1 - need for current milestone

### Impact

This can cause frustration with keyboard users when they are using `Escape` to step back into the previous expected focus-trap.

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**Comment by timmorey** [comment]
I found this while searching issues before reporting a bug, and it sounds like this may capture the bug I was going to report.

We found that when a `<calcite-popover closable>` is nested within a `<calcite-panel closable>`, and focus is in the open popover, hitting `Esc` will close both the popover and the panel.  It feels like it should only close the popover.  Codepen: https://codepen.io/_timmorey_/pen/emZwqjm?editors=100

@jcfranco Should I report a new bug, or is this covered by the current issue?

**Comment by jcfranco** [comment]
@timmorey It should be covered by this one. Thanks for the ping!

**Comment by cont-egorova** [comment]
I'm experiencing similar behavior.

When a popover has an open tooltip, pressing Esc closes both the popover and the tooltip. It is expected that Esc should close only the tooltip, leaving the popover open.

Codepen: https://codepen.io/natalia_eg/pen/qENaLOB

@jcfranco Should I log this as a separate issue, or does the existing one cover it?

**Comment by jcfranco** [comment]
@cont-egorova I believe it should also be covered. If not, we'll create a follow-up issue. Thanks for sharing your use case!

**Comment by jcfranco** [comment]
@Esri/calcite-pes We should consider adding this to an upcoming milestone.

**Comment by jcfranco** [comment]
Created the following issue: https://github.com/focus-trap/focus-trap/issues/1846

**Comment by jcfranco** [comment]
> Created the following issue: [focus-trap/focus-trap#1846](https://github.com/focus-trap/focus-trap/issues/1846)

I'll proceed with a suggestion from ☝️. We can unblock this and adjust if there are any upstream changes down the road.

**Comment by DitwanP** [comment]
🍠 Verified on `5.1.0-next.43`, escape closes opened components one at a time in reverse order of opening.

https://codepen.io/Ditwan-Price/pen/GgNRdaJ

**Comment by minhiu** [comment]
> I'm experiencing similar behavior.
> 
> When a popover has an open tooltip, pressing Esc closes both the popover and the tooltip. It is expected that Esc should close only the tooltip, leaving the popover open.
> 
> Codepen: https://codepen.io/natalia_eg/pen/qENaLOB
> 
> [@jcfranco](https://github.com/jcfranco) Should I log this as a separate issue, or does the existing one cover it?

@jcfranco Did we also fix for this case where @cont-egorova brought up?

I was still able to reproduce the issue here

https://codepen.io/Hieu-Pham-the-animator/pen/JoEXGbr

**Comment by jcfranco** [comment]
@minhiu Good callout. That use case is indeed a separate issue. I’ll look into it. 🔍

@DitwanP, could you help create an issue to track that use case?
